scriptworker/github.py

```diff
@@ -65,3 +76,152 @@ def get_release(self, tag_name):
 
         """
         return self._github_repository.release_from_tag(tag_name).as_dict()
+
+    def get_tag_hash(self, tag_name):
+        """Fetch the commit hash that was tagged with ``tag_name``.
+
+        Args:
+            tag_name (str): the name of the tag
+
+        Returns:
+            str: the commit hash linked by the tag
+
+        """
+        tag_object = get_single_item_from_sequence(
+            sequence=self._github_repository.tags(),
+            condition=lambda tag: tag.name == tag_name,
+            no_item_error_message='No tag "{}" exist'.format(tag_name),
+            too_many_item_error_message='Too many tags "{}" found'.format(tag_name),
+        )
+
+        return tag_object.commit.sha
+
+    async def has_commit_landed_on_repository(self, context, revision):
+        """Tell if a commit was landed on the repository or if it just comes from a pull request.
+
+        Args:
+            context (scriptworker.context.Context): the scriptworker context.
+            revision (str): the commit hash or the tag name.
+
+        Returns:
+            bool: True if the commit is present in one of the branches of the main repository
+
+        """
+        # Revision may be a tag name. `branch_commits` doesn't work on tags
+        if not _is_git_full_hash(commit_hash=revision):
+            revision = self.get_tag_hash(tag_name=revision)
```

Comment:
At the moment, that's not a supported case. `get_tag_hash()` will raise an error because the tag doesn't exist. I'm not aware of a case where we want to support it. If that comes up, I'm fine adding this support.